src/Razor/src/Microsoft.AspNetCore.Razor.Language/Extensions/AttributeDirective.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language.Extensions
+{
+    internal static class AttributeDirective
+    {
+        public static readonly DirectiveDescriptor Directive = DirectiveDescriptor.CreateDirective(
+            "attribute",
+            DirectiveKind.SingleLine,
+            builder =>
+            {
+                builder.AddAttributeToken(ComponentResources.AttributeDirective_AttributeToken_Name, ComponentResources.AttributeDirective_AttributeToken_Description);
+                builder.Usage = DirectiveUsage.FileScopedMultipleOccurring;
+                builder.Description = ComponentResources.AttributeDirective_Description;
+            });
+
+        public static void Register(RazorProjectEngineBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.AddDirective(Directive, FileKinds.Legacy, FileKinds.Component, FileKinds.ComponentImport);
```

Comment:
I assume we want this to work on non-blazor scenarios as well. I don't see why not.

Reply:
Yes we do.